source/_integrations/template.markdown

```diff
@@ -112,12 +112,12 @@ sensor:
       required: true
       type: template
     unit_of_measurement:
-      description: "Defines the units of measurement of the sensor, if any. This will also influence the graphical presentation in the history visualization as a continuous value. Sensors with missing `unit_of_measurement` are showing as discrete values."
+      description: "Defines the units of measurement of the sensor, if any. This will also display the value based on the user profile Number Format setting and influence the graphical presentation in the history visualization as a continuous value. Sensors with missing `unit_of_measurement` are showing as discrete values."
       required: false
       type: string
       default: None
     state_class:
-      description: The [state_class](https://developers.home-assistant.io/docs/core/entity/sensor#available-state-classes) of the sensor.
+      description: "The [state_class](https://developers.home-assistant.io/docs/core/entity/sensor#available-state-classes) of the sensor. Only possible value currently is `measurement`. Set this if your template sensor represents a measurement of the current value (so not a daily aggregate etc). This will also display the value based on the user profile Number Format setting and influence the graphical presentation in the history visualization as a continuous value." 
```

Comment:
This reverts changes from #20243, please fix that.
Maybe mention that if neither `state_class` nor `unit_of_measurement` is defined, the history view will display discrete values.

Reply:
I revised the description so that `unit_of_measurement` and `state_class` mention how they affect Number Format and history visualization without the previous description of `state_class`.

The legacy style sensor can not use `state_class` (throws an error when checking the configuration) so I did not mention it in that description of the `unit_of_measurement`. I revised that description to remove the last sentence that explicitly states it will be shown as discrete values; now it is consistent with the descriptions in the current method to configure template sensors.